src/content/reference/react/cache.md

```diff
@@ -0,0 +1,485 @@
+---
+title: cache
+canary: true
+---
+
+<Canary>
+* `cache` is only for use with [React Server Components](https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components). See [frameworks](https://react.dev/learn/start-a-new-react-project#bleeding-edge-react-frameworks) that support React Server Components.
+
+* `cache` is currently only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachedFn = cache(fn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cache(fn)` {/*cache*/}
+
+Call `cache` outside of any components to create a version of the function with caching.
+
+```js {4,7}
+import {cache} from 'react';
+import calculateMetrics from 'lib/metrics';
+
+const getMetrics = cache(calculateMetrics);
+
+function Chart({data}) {
+  const report = getMetrics(data);
+  // ...
+}
+```
+
+`getMetrics` will first check if there is a cached result for `data`, otherwise call `calculateMetrics(data)` and cache the returned result.
```

Comment:
Updated to 

```
When `getMetrics` is first called with `data`, `getMetrics` will call `calculateMetrics(data)` and store the result in cache. If `getMetrics` is called again with the same `data`, it will return the cached result instead of calling `calculateMetrics(data)` again.
```